manifests/monitoring/observability/thanosCompactor.yaml

```diff
@@ -0,0 +1,72 @@
+apiVersion: apps/v1
+kind: StatefulSet
```

Comment:
Why does this need to be a statefulset ?

Reply:
Because Compactor by itself is not concurrency safe:
https://github.com/thanos-io/thanos/blob/master/docs/components/compact.md

Reply:
I think a Deployment with a maxSurge: 0, and maxUnavailable: 1 is a better way to model this constraint.